lib/hashicorptools/code_deploy.rb

```diff
@@ -33,36 +76,36 @@ def deploy
                  g.log.first
                end
 
+      puts "Deploying to environment #{options[:environment]} - regions: #{aws_regions.join(', ')}
+            commit: #{commit.sha}
+            message: #{commit.message}"
 
-      puts "deploying commit: #{commit.sha} #{commit.message}"
+      puts "Deploying for regions: #{aws_regions}"
 
-      create_deployment(commit.sha, commit.message)
+      threads = []
+      aws_regions.each_slice(2) do |aws_regions_batch|
+        puts "Deploying for 2 regions: #{aws_regions_batch}"
+        aws_regions_batch.each do |aws_region|
+          thread = Thread.new{ region_deployment(aws_region).create_deployment(commit.sha, commit.message) }
+          threads.push(thread)
+          begin
+            thread.join
```

Comment:
I think this will block the main thread and avoid the second thread to be run until the first one finishes. 
Instead we should do the `.join` call outside the loop  with `threads.map {|thread| thread.join}` (or with the exception handling inside the `map` block)